public/js/components/FormFields/FormFieldArrayWrapper.js

```diff
@@ -18,27 +18,51 @@ export default class FormFieldArrayWrapper extends React.Component {
     ])
   }
 
+  //We store new items in state to avoid sending invalid items back to the server before they're ready
+  state = {
+    newItems: []
+  }
+
   onAddClick = () => {
-    const existingValues = this.props.fieldValue || [];
-    this.props.onUpdateField(existingValues.concat([undefined]));
+    this.setState({
+      newItems: this.state.newItems.concat([undefined])
+    });
   }
 
   renderValue(value, i) {
 
     const updateFn = (newValue) => {
-      //Find the value in the array to change
-      const newFieldValue = this.props.fieldValue.map((oldValue) => {
-        return value === oldValue ? newValue : oldValue;
-      });
-      this.props.onUpdateField(newFieldValue);
+      if (this.state.newItems.indexOf(value) > -1) {
```

Comment:
`if(this.state.newItems.includes(value))`

Reply:
Also you have access to `i` from `renderValue`, so in theory you shouldn't need to do this. You can just then test if `this.state.newItems[i] === undefined`.

Reply:
hmm, `i` is the index in the concatenation of `this.props.fieldValue` and `this.state.newItems`.
In fact, I've just tried this with `if (value === undefined)` and it works. This is quite nasty